### PR TITLE
tests: fix parametrization

### PR DESCRIPTION
**Pull request by dtrifiro**
By removing autouse in the `connection_type` fixture, we will only parametrize over tests which use `connection_type`. Since the client fixtures use connection_type, we're automatically parametrizing for all client tests (and skipping the ones which do not need parametrizing)

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -37,7 +37,6 @@ def model_name(request: pytest.FixtureRequest):
 
 
 @pytest.fixture(
-    autouse=True,
```
Comment (z103cb) [inline tests/conftest.py:40]: Why was this necessary ?

**Review by z103cb — APPROVED** [review]
LGTM, I am not sure about the code coverage issue.

File: tests/conftest.py (hunk 1 above)
Comment (dtrifiro) [reply tests/conftest.py:40]: `autouse` makes all tests use the fixture, even when not requested, we added it at the beginning thinking that all tests would require this, but it's actually only needed for some tests, thus we can relax this.

**Comment by dtrifiro** [comment]
Merge after #71
